### PR TITLE
The config.json for each run is now saved inside the logs folder

### PR DESCRIPTION
**Pull request by VFjr**
This is related to backend part of Issue #213 

Now every time you run a simulation, the config.json file for that simulation will be saved in the logs folder together with the other logs. This will allow the frontend to pull and display the configs for each run afterwards

### CODE REVIEW

**Review by JaafarRammal — COMMENTED** [review]
LGTM, waiting for non-infra approval

**Review by jzzheng22 — APPROVED** [review]
sure i trust

**Pull request merged**
